src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -909,6 +909,15 @@ internal int CalculateSyntaxOffsetInSynthesizedConstructor(int position, SyntaxT
                 return syntaxOffset;
             }
 
+            if (localKind == SynthesizedLocalKind.InstrumentationPayload)
```

Comment:
>localKind == SynthesizedLocalKind.InstrumentationPayload [](start = 16, length = 56)

How about changing the condition to 
position == declaration.Location.SpanStart? That should express the fact that we are asking for offset to variable/closure scope with syntax == the declaring type syntax.